plugins/org.jkiss.dbeaver.ui.editors.data/src/org/jkiss/dbeaver/ui/controls/lightgrid/GridColumn.java

```diff
@@ -214,18 +215,32 @@ int computeHeaderWidth() {
         }
         {
             int textWidth;
-            if (Boolean.TRUE.equals(labelProvider.getGridOption(IGridLabelProvider.OPTION_EXCLUDE_COLUMN_NAME_FOR_WIDTH_CALC))) {
-                textWidth = grid.sizingGC.stringExtent("X").x;
-            } else {
-                String text = labelProvider.getText(this);
-                String description = labelProvider.getDescription(this);
-                textWidth = grid.sizingGC.stringExtent(text).x;
-                if (!CommonUtils.isEmpty(description)) {
-                    int descWidth = grid.sizingGC.stringExtent(description).x;
-                    if (descWidth > textWidth) {
-                        textWidth = descWidth;
+            Object calcWidthMethod = labelProvider.getGridOption(IGridLabelProvider.OPTION_CALC_COLUMN_WIDTH_METHOD);
+            String text = "X";
+            switch ((ResultSetPreferences.GridColumnCalcWidthMethod) calcWidthMethod) {
+                case TITLE_AND_VALUES:
+                    text = labelProvider.getText(this);
+                    textWidth = grid.sizingGC.stringExtent(text).x;
+                    break;
+                case VALUES:
+                    textWidth = grid.sizingGC.stringExtent(text).x;
+                    break;
+                case TITLE_DESCRIPTION_VALUES:
+                    text = labelProvider.getText(this);
+                    textWidth = grid.sizingGC.stringExtent(text).x;
+                    if (Boolean.TRUE.equals(labelProvider.getGridOption(IGridLabelProvider.OPTION_SHOW_DESCRIPTION))) {
+                        String description = labelProvider.getDescription(this);
+                        if (!CommonUtils.isEmpty(description)) {
+                            int descWidth = grid.sizingGC.stringExtent(description).x;
+                            if (descWidth > textWidth) {
+                                textWidth = descWidth;
+                            }
+                        }
                     }
-                }
+                    break;
+                default:
+                    textWidth = grid.sizingGC.stringExtent("X").x;
+                    break;
```

Comment:
It should not be required if the `switch` statement covers all enum variants. If you really want to, make one of the variants to be `default`.